lib/control-connection.js

```diff
@@ -429,40 +428,93 @@ ControlConnection.prototype.nodeStatusChangeHandler = function (event) {
  */
 ControlConnection.prototype.nodeSchemaChangeHandler = function (event) {
   this.log('info', 'Schema change', event);
-  if (event.schemaChangeType !== schemaChangeTypes.created) {
-    var ksInfo = this.metadata.keyspaces[event.keyspace];
-    if (!ksInfo) {
-      //It hasn't been loaded and it is not part of the metadata, don't mind
-      return;
+  if (!this.options.isMetadataEnabled) {
+    return;
+  }
+  this.handleSchemaChange(event);
+};
+
+/** @param {{keyspace: string, isKeyspace: boolean, schemaChangeType, table, udt, functionName, aggregate}} event */
+ControlConnection.prototype.handleSchemaChange = function (event) {
+  var self = this;
+  var handler, cqlObject;
+  if (event.isKeyspace) {
+    if (event.schemaChangeType === schemaChangeTypes.dropped) {
+      handler = function removeKeyspace() {
+        // if on the same event queue there is a creation, this handler is not going to be executed
+        // it is safe to remove the keyspace metadata
+        delete self.metadata.keyspaces[event.keyspace];
+      };
+      return this.scheduleObjectRefresh(handler, event.keyspace, null);
```

Comment:
Just curious if there is a difference between doing scheduleObjectRefresh and scheduleKeyspaceRefresh below?  Might be some small detail i'm missing.